plugins/modules/azure_rm_recoveryservicesvault.py

```diff
@@ -0,0 +1,240 @@
+            resource_group=dict(
+                type='str',
+                required=True
+            ),
+            name=dict(
+                type='str',
+                required=True
+            ),
+            location=dict(
+                type='str',
+                required=True
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+        self.state = None
+
+        self.results = dict(changed=False)
+        self.mgmt_client = None
+        self.url = None
+        self.status_code = [200, 201, 202, 204]
+
+        self.body = {}
+        self.query_parameters = {}
+        self.query_parameters['api-version'] = None
+        self.header_parameters = {}
+        self.header_parameters['Content-Type'] = 'application/json; charset=utf-8'
+
+        super(AzureRMRecoveryServicesVault, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                            supports_check_mode=True,
+                                            supports_tags=True)
+
+    def get_api_version(self):
+        return '2016-06-01' if self.state == 'present' or self.state == 'absent' else '2019-05-13'
+
+    def get_url(self):
+        if self.state == 'present' or self.state == 'absent':
+            return '/subscriptions/' \
+                   + self.subscription_id \
+                   + '/resourceGroups/' \
+                   + self.resource_group \
+                   + '/providers/Microsoft.RecoveryServices' \
+                   + '/vaults' + '/' \
+                   + self.name
+
+    def get_body(self):
+        if self.state == 'present':
+            return {
+                "properties": {},
+                "sku": {
+                    "name": "Standard"  #The SKU is always "Standard".
+                    },
+                "location": self.location
+            }
+        else:
+            return {}
+
+    def exec_module(self, **kwargs):
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                self.body[key] = kwargs[key]
+
+        self.inflate_parameters(self.module_arg_spec, self.body, 0)
+
+        self.query_parameters['api-version'] = self.get_api_version()
+        self.url = self.get_url()
+        self.body = self.get_body()
+        old_response = None
+        response = None
+
+        self.mgmt_client = self.get_mgmt_svc_client(GenericRestClient,
+                                                    base_url=self._cloud_environment.endpoints.resource_manager)
+
+        changed = False
+        if self.state == 'present':
+            changed = True
+            response = self.create_recovery_service_vault()
```

Comment:
Hi @suyeb786 Under review! Can we read the recover vault information here, and then determine whether it exists, if it exists, do not update, if it does not exist, create it! Thank you!

Reply:
Hi @Fred-sun,
Changes are completed.